sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/expressions/ObjectExpressionsSuite.scala

```diff
@@ -202,6 +203,20 @@ class ObjectExpressionsSuite extends SparkFunSuite with ExpressionEvalHelper {
       Invoke(funcSubObj, "binOp", DoubleType, inputSum), 0.75, InternalRow.apply(1, 0.25))
   }
 
+  test("SPARK-31854: Invoke in MapElementsExec should not propagate null") {
+    val targetObject = new InvokeTargetClass
+    val funcClass = classOf[InvokeTargetClass]
+    val funcObj = Literal.create(targetObject, ObjectType(funcClass))
+    val inputInt = Seq(BoundReference(0, ObjectType(classOf[Any]), true))
+    val outputType = ObjectType(classOf[(Any, Any)])
+    val inputRow = InternalRow.fromSeq(Seq(null.asInstanceOf[java.lang.Integer]))
+    val createExpr = (propagateNull: Boolean) => {
+      Invoke(funcObj, "mapFunc", outputType, inputInt, propagateNull)
+    }
+    checkObjectExprEvaluation(createExpr(true), null, inputRow)
+    checkObjectExprEvaluation(createExpr(false), (null, null), inputRow)
+  }
+
```

Comment:
I think this test is already passed before this change, right?

Reply:
Ah, yes.